File: problems/0965-unique-email-addresses/solution.cpp
#include <set>
// solution in 15 mins
class Solution {
public:
    int numUniqueEmails(vector<string>& emails) {
        set<string> uniqueEmails;
        for (string e : emails) {
            string name;
            for (char c : e) {
                if (c == '@' || c == '+') break;
                if (c == '.') continue;
                name += c;
            }
            string domain = e.substr(e.find('@'));
            name += domain;
            uniqueEmails.insert((name));
        }
        return uniqueEmails.size();
    }
};

File: problems/0965-unique-email-addresses/solution.py
class Solution:
    def numUniqueEmails(self, emails: List[str]) -> int:
        eset = set()
        count = 0

        for e in emails:
            parts = e.split("@")
            name = parts[0]
            domain = parts[-1]
            
            if "+" in name:
                name = name.split("+")[0]
            name = name.replace(".", "")

            if (name, domain) not in eset:
                count += 1
                eset.add((name, domain))

        return count
